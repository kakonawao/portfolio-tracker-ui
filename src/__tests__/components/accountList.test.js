import React from "react";
import { shallow } from "enzyme";

import AccountList from "../../components/accountsList";


describe("<AccountList>", () => {
    it("renders mesage when data is loading", () => {
        const props = {
            data: {
                loading: true,
                items: []
            }
        };

        const wrapper = shallow(<AccountList {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders empty list", () => {
        const props = {
            data: {
                loading: false,
                items: []
            }
        };

        const wrapper = shallow(<AccountList {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders list when items are present", () => {
        const props = {
            data: {
                loading: false,
                items: [
                    {
                        code: "1",
                        description: "bank account",
                        holder: {
                            name: "Bank of Ireland"
                        }
                    },
                    {
                        code: "2",
                        description: "cash account",
                        holder: null
                    }
                ]
            }
        };

        const wrapper = shallow(<AccountList {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
