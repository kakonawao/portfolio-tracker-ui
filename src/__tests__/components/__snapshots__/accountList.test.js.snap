// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountList> renders empty list 1`] = `
<p>
  You have no accounts.
</p>
`;

exports[`<AccountList> renders list when items are present 1`] = `
Array [
  <AccountItem
    code="1"
    description="bank account"
    holder={
      Object {
        "name": "Bank of Ireland",
      }
    }
    key="1"
  />,
  <AccountItem
    code="2"
    description="cash account"
    holder={null}
    key="2"
  />,
]
`;

exports[`<AccountList> renders mesage when data is loading 1`] = `
<p>
  Loading accounts...
</p>
`;
